### PR TITLE
Perception: fix detection component process error

### PR DESCRIPTION
**Pull request by jeroldchen**

### CODE REVIEW

File: modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h [hunk 1]
```diff
@@ -42,6 +42,8 @@ class PointPillarsDetection {
 
   bool Detect(const DetectionOptions& options, LidarFrame* frame);
 
+  std::string Name() { return "PointPillarsDetection"; }
```
Comment (jinghaomiao) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h:45]: Use a static const variable or a gflag for the string "PointPillarsDetection".

File: modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h [hunk 2]
```diff
@@ -55,15 +57,21 @@ class PointPillarsDetection {
...
   float score_threshold_ = 0.5;
   float nms_overlap_threshold_ = 0.5;
-  std::string pfe_onnx_file_;
-  std::string rpn_onnx_file_;
+  std::string pfe_onnx_file_ =
+      "/apollo/modules/perception/lidar/lib/detection/lidar_point_pillars/"
+      "test/data/pfe.onnx";
+  std::string rpn_onnx_file_ =
+      "/apollo/modules/perception/lidar/lib/detection/lidar_point_pillars/"
+      "test/data/rpn.onnx";
```
Comment (jinghaomiao) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h:75]: If the two files are not for testing, move them out of test folder.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h (hunk 1 above)
Comment (lx18233184051) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h:45]: I think so, avoid return a string var directly.

**Comment by lx18233184051** [comment]
approve

File: modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h (hunk 1 above)
Comment (jeroldchen) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h]: Have made it const.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h (hunk 2 above)
Comment (jeroldchen) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/include/point_pillars_detection.h:76]: Have copied them into perception/production/data/models, but they need updated to a better trained version soon. And original files are still requested by the unit test.

File: modules/perception/lidar/app/BUILD [hunk 3]
```diff
@@ -22,13 +22,13 @@ cc_library(
...
         "//modules/perception/lib/utils",
         "//modules/perception/lidar/app/proto:app_proto",
+        "//modules/perception/lidar/lib/detection/lidar_point_pillars:point_pillars_detection",
         "//modules/perception/lidar/lib/map_manager",
         "//modules/perception/lidar/lib/object_builder",
         "//modules/perception/lidar/lib/object_filter_bank",
         "//modules/perception/lidar/lib/pointcloud_preprocessor",
         "//modules/perception/lidar/lib/scene_manager",
         "//modules/perception/proto:perception_config_schema_proto",
-        "//modules/perception/lidar/lib/detection/lidar_point_pillars:point_pillars_detection",
```
Comment (lx18233184051) [inline modules/perception/lidar/app/BUILD:31]: 这个文件是不是其实没有更改？如果没有改，下次就不用提了

Comment (jeroldchen) [reply modules/perception/lidar/app/BUILD:31]: Just keep the items in alphabetical order.

File: modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc [hunk 4]
```diff
@@ -139,8 +139,8 @@ TestClass::TestClass(const int MAX_NUM_PILLARS,
   float nms_overlap_threshold = 0.5;
   std::string package_path =
       "/apollo/modules/perception/lidar/lib/detection/lidar_point_pillars";
-  std::string pfe_onnx_path = "/test/data/pfe.onnx";
-  std::string rpn_onnx_path = "/test/data/rpn.onnx";
+  std::string pfe_onnx_path = "/data/pfe.onnx";
+  std::string rpn_onnx_path = "/data/rpn.onnx";
   std::string pfe_onnx_file = package_path + pfe_onnx_path;
   std::string rpn_onnx_file = package_path + rpn_onnx_path;
```
Comment (jinghaomiao) [inline modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc:146]: 建议使用Gflag来传达model file，可以参考predcition部分代码：https://github.com/ApolloAuto/apollo/blob/master/modules/prediction/evaluator/vehicle/junction_map_evaluator.cc#L189

Comment (jeroldchen) [reply modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_test.cc:141]: OK, I have moved the definition of these variables into perception_gflags.

**Pull request merged**